### PR TITLE
Make it clearer that unimplemented callbacks always stop

### PR DESCRIPTION
**Pull request by lexmag**
\cc @fishcakez

### CODE REVIEW

File: lib/elixir/lib/gen_server.ex [hunk 1]
```diff
@@ -409,9 +409,10 @@ defmodule GenServer do
       @doc false
       def handle_call(msg, _from, state) do
         # We do this to trick Dialyzer to not complain about non-local returns.
-        case :random.uniform(1) do
-          1 -> exit({:bad_call, msg})
-          2 -> {:noreply, state}
```
Comment (fishcakez) [inline lib/elixir/lib/gen_server.ex]: ergh, :noreply here, that is confusing

File: lib/elixir/lib/gen_event/stream.ex [hunk 2]
```diff
@@ -28,9 +28,10 @@ defmodule GenEvent.Stream do
   @doc false
   def handle_call(msg, _state) do
     # We do this to trick dialyzer to not complain about non-local returns.
-    case :random.uniform(1) do
-      1 -> exit({:bad_call, msg})
-      2 -> {:remove_handler, :ok}
+    reason = {:bad_call, msg}
+    case :erlang.phash2(1, 1) do
```
Comment (fishcakez) [inline lib/elixir/lib/gen_event/stream.ex]: We should do this for handle_event here too.

**Comment by fishcakez** [comment]
Maybe this is bike shedding but in `Connection` switched to use a persistent hash because it was clearer it is always the same value - rather than a random number. This is also side effect free.

File: lib/elixir/lib/gen_event/stream.ex (hunk 2 above)
Comment (lexmag) [inline lib/elixir/lib/gen_event/stream.ex]: Sorry, I didn't get what "this"?

Comment (lexmag) [inline lib/elixir/lib/gen_event/stream.ex]: Do you mean `exit` by default?

Comment (fishcakez) [inline lib/elixir/lib/gen_event/stream.ex]: We don't use the trick on `handle_event/2` in this file, just `exit/1`.

Comment (lexmag) [inline lib/elixir/lib/gen_event/stream.ex]: Added, I was looking in the `gen_event.ex` file, of course. :see_no_evil:

**Comment by fishcakez** [comment]
Thank you!

**Comment by lexmag** [comment]
Glad to help. :)

**Pull request merged**